power_grid_model_c/power_grid_model/include/power_grid_model/main_model_impl.hpp

```diff
@@ -218,6 +184,18 @@ class MainModelImpl<ExtraRetrievableTypes<ExtraRetrievableType...>, ComponentLis
         return result;
     }
 
+    // helper function to get the number of components per type
+    std::array<Idx, n_types> get_n_components_per_type() const {
+        std::array<Idx, n_types> result{};
+        size_t idx{};
+        main_core::utils::run_functor_with_all_types_return_void<ComponentType...>(
+            [&result, this, &idx]<typename CompType>() {
+                result[idx] = this->component_count<CompType>();
+                ++idx;
```

Comment:
I know this is against all 'best practices' and 'recommendations', but just for the beauty:
```suggestion
                result[idx++] = this->component_count<CompType>();
```

Reply:
See other comment.